src/watch-run.ts

```diff
@@ -37,26 +45,60 @@ export function makeWatchRun(instance: TSInstance) {
           filePath.match(constants.dtsDtsxOrDtsDtsxMapRegex) !== null &&
           filePath.match(constants.nodeModules) === null
         ) {
-          updateFile(instance, filePath);
+          promises.push(updateFile(instance, filePath, loader, loaderIndex));
         }
       }
     }
 
     // Update all the watched files from solution builder
     if (instance.solutionBuilderHost) {
       for (const filePath of instance.solutionBuilderHost.watchedFiles.keys()) {
-        updateFile(instance, filePath);
+        promises.push(updateFile(instance, filePath, loader, loaderIndex));
       }
     }
 
-    callback();
+    Promise.all(promises)
```

Comment:
I can't remember if we've discussed this already, are there any issues with us moving to a `Promise` (ie async) approach instead of the current approach? Given we're dealing with callbacks I suspect not, but I wanted to check

Reply:
I think there are no new issues introduced by the 'async' approach.